controller/watcher.go

```diff
@@ -46,11 +50,33 @@ func newWatcher(conf *config.Config) (*hcat.Watcher, error) {
 	}
 
 	return hcat.NewWatcher(hcat.WatcherInput{
-		Clients: clients,
-		Cache:   hcat.NewStore(),
+		Clients:         clients,
+		Cache:           hcat.NewStore(),
+		ConsulRetryFunc: retryConsul,
 	}), nil
 }
 
+// retryConsul will be used by hashicat watcher to retry polling Consul for
+// catalog changes. If polling Consul fails after retries, CTS will actually
+// exit.
+//
+// retryCount parameter is passed in by hcat. It starts at 0 (there have been
+// zero retries).
+func retryConsul(retryCount int) (bool, time.Duration) {
+	max := 8 // 8+1 retries. total wait of 8.5-12.8 min
+	if retryCount > max {
+		log.Printf("[ERR] (hcat) error connecting with Consul even after retries")
+		return false, 0 * time.Second
+	}
+
+	r := retry.NewRetry(0, time.Now().UnixNano()) // 0 is not used
+	wait := r.WaitTime(uint(retryCount))
+	dur := time.Duration(wait)
```

Comment:
This is a round-about way to get the exponential backoff duration from the retry count. What do you think about refactoring `WaitTime` to be a standalone function in addition to the retry object's method? Something like:


```go
func WaitTime(attempt uint, random) int {
  // move current r.waitTime logic here
}

func (r Retry) waitTime(attempt uint) int {
	if r.testMode {
		return 1
	}
	return WaitTime(attempt, r.random.Float64())
}
```

Reply:
Good idea! I like this a lot. It's a lot cleaner. I will make this refactor!